yt_dlp/extractor/tiktok.py

```diff
@@ -793,102 +792,150 @@ def _real_extract(self, url):
+                query=self._build_web_query(sec_uid, cursor), headers={'User-Agent': self._USER_AGENT})
+
+            for video in traverse_obj(response, ('itemList', lambda _, v: v['id'])):
+                video_id = video['id']
+                webpage_url = self._create_url(user_name, video_id)
+                info = try_call(
+                    lambda: self._parse_aweme_video_web(video, webpage_url, video_id)) or {'id': video_id}
+                info.pop('formats', None)
+                yield self.url_result(webpage_url, TikTokIE, **info)
+
+            old_cursor = cursor
+            cursor = traverse_obj(
+                response, ('itemList', -1, 'createTime', {lambda x: x * 1E3}, {int_or_none}))
+            if not cursor:
+                cursor = old_cursor - 604800000  # jump 1 week back in time
+            if cursor < 1472706000000 or not traverse_obj(response, 'hasMorePrevious'):
                 break
-            query['max_cursor'] = post_list['max_cursor']
-
-    def _entries_api(self, user_id, videos):
-        for video in videos:
-            yield {
-                **self._parse_aweme_video_app(video),
-                'extractor_key': TikTokIE.ie_key(),
-                'extractor': 'TikTok',
-                'webpage_url': f'https://tiktok.com/@{user_id}/video/{video["aweme_id"]}',
-            }
+
+    def _get_sec_uid(self, user_url, user_name, msg):
+        webpage = self._download_webpage(
+            user_url, user_name, fatal=False, headers={'User-Agent': 'Mozilla/5.0'},
+            note=f'Downloading {msg} webpage', errnote=f'Unable to download {msg} webpage') or ''
+        return traverse_obj(
+            self._get_universal_data(webpage, user_name),
+            ('webapp.user-detail', 'userInfo', 'user', 'secUid', {str})) or traverse_obj(
+            self._get_sigi_state(webpage, user_name),
+            ('LiveRoom', 'liveRoomUserInfo', 'user', 'secUid'),
+            ('UserModule', 'users', ..., 'secUid'),
+            get_all=False, expected_type=str)
 
     def _real_extract(self, url):
-        user_name = self._match_id(url)
-        webpage = self._download_webpage(url, user_name, headers={
-            'User-Agent': 'facebookexternalhit/1.1 (+http://www.facebook.com/externalhit_uatext.php)'
-        })
-        user_id = self._html_search_regex(r'snssdk\d*://user/profile/(\d+)', webpage, 'user ID', default=None) or user_name
+        user_name, sec_uid = None, None
+        if url.startswith('tiktokuser:'):
+            sec_uid = self._match_id(url)
+        else:
+            user_name = self._match_id(url)
+
+        if not sec_uid:
+            for user_url, msg in (
+                (self._UPLOADER_URL_FORMAT % user_name, 'user'),
+                (self._UPLOADER_URL_FORMAT % f'{user_name}/live', 'live'),
+            ):
+                sec_uid = self._get_sec_uid(user_url, user_name, msg)
+                if sec_uid:
+                    break
```

Comment:
```suggestion
        for user_url, msg in (
            (self._UPLOADER_URL_FORMAT % user_name, 'user'),
            (self._UPLOADER_URL_FORMAT % f'{user_name}/live', 'live'),
        ):
            if sec_uid:
                break
            sec_uid = self._get_sec_uid(user_url, user_name, msg)
```

Reply:
applied, but then had to revert/change due to change in user_name vs sec_uid logic